datahub-upgrade/src/main/java/com/linkedin/datahub/upgrade/restorebackup/backupreader/BackupReader.java

```diff
@@ -0,0 +1,14 @@
+package com.linkedin.datahub.upgrade.restorebackup.backupreader;
+
+import com.linkedin.datahub.upgrade.UpgradeContext;
+
+
+/**
+ * Base interface for BackupReader used for creating the BackupIterator to retrieve EbeanAspectV2 object to be
+ * ingested back into GMS
+ */
+public interface BackupReader {
+  String getName();
+
+  EbeanAspectBackupIterator getBackupIterator(UpgradeContext context);
```

Comment:
Again, I wish this interface didn't have to know about UpgradeContext

Reply:
This was mainly because of the logging. If we can just log.info, then we don't need to pass in the context.

Reply:
forgot about this -- we should be able to just use the logger as well